File: [프로그래머스] 2017 팁스타운_짝지어 제거하기.py
def solution(s):
    answer = []
    for i in s:
        if len(answer) == 0:    #비어있을 때
            answer.append(i)
        elif answer[-1] == i:   #top과 s의 문자가 동일할 때
            answer.pop()    #answer top 제거
        else:
            answer.append(i)
    if len(answer) == 0:    #answer이 비어 있으면 정상 수행 완료
        return 1
    else:
        return 0

File: [프로그래머스] 2017 팁스타운_짝지어 제거하기.js
function solution(s)
{
    let stack = [];
    
    for(let i of s){
        if(!stack){
            stack.push(i);
        } else {
            if(stack[stack.length - 1] === i){
                stack.pop();
            } else {
                stack.push(i)
            }
        }
    }
    
    return stack.length ? 0 : 1;
}